framework/quota/src/main/java/org/apache/cloudstack/quota/QuotaManagerImpl.java

```diff
@@ -133,9 +134,21 @@ public boolean stop() {
         return true;
     }
 
+    public boolean isQuotaEnabledForAccountDomain(final AccountVO account) {
+        boolean isQuotaAccountEnabled = QuotaConfig.QuotaAccountEnabled.valueIn(account.getAccountId());
+        if (!isQuotaAccountEnabled) {
+            s_logger.debug(String.format("Considering usage records as calculated and skipping it because the account [%s] has the quota plugin disabled.", account));
+        }
+        return isQuotaAccountEnabled;
+    }
+
+    public boolean isUsageRecordsEmpty(final Pair<List<? extends UsageVO>, Integer> usageRecords) {
+        return usageRecords == null || usageRecords.first() == null || usageRecords.first().isEmpty();
```

Comment:
We can use `org.apache.commons.collections.CollectionUtils#isEmpty` here.
```suggestion
        return usageRecords == null || CollectionUtils.isEmpty(usageRecords.first());
```